### PR TITLE
correlation id

### PR DESCRIPTION
**Pull request by robertstriz**
## Summary

🚨 Provide a brief summary of your work.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactoring (non-breaking change to code and structure)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Testing

🚨 Provide info about the new tests and tests result

## Notes

🚨 Anything else

### CODE REVIEW

**Review by PlesnikJakub — APPROVED** [review]

File: src/WebApi/Logging/TimeElapsedDiagnosticsMiddleware.cs [hunk 1]
```diff
@@ -17,19 +19,36 @@ public TimeElapsedDiagnosticsMiddleware(RequestDelegate next)
...
         }
     }
 
+    private static string GetCorrelationId(HttpContext context)
+    {
+        context.Request.Headers.TryGetValue("correlation-id", out StringValues correlationIds);
+        return correlationIds.FirstOrDefault() ?? Guid.NewGuid().ToString();
+    }
+    
+    private static void SetCorrelationId(string correlationId,HttpContext context)
```
Comment (petrspelos) [inline src/WebApi/Logging/TimeElapsedDiagnosticsMiddleware.cs:47]:
```suggestion
    private static void SetCorrelationId(string correlationId, HttpContext context)
```

trivial lint

File: src/WebApi/Logging/TimeElapsedDiagnosticsMiddleware.cs [hunk 2]
```diff
@@ -17,19 +19,36 @@ public TimeElapsedDiagnosticsMiddleware(RequestDelegate next)
...
-        }
-        catch (Exception e)
-        {
-            PushPropertyElapsed(sw);
-            Log.Fatal(e,"TimeElapsedDiagnosticsMiddleware - ERROR Message: {Message}", e.Message);
+            try
+            {
+                await _next.Invoke(context);
+                PushPropertyElapsed(sw);
+                Log.Information("TimeElapsedDiagnosticsMiddleware - OK - CorrelationId: {CorrelationId}",correlationId);
```
Comment (petrspelos) [inline src/WebApi/Logging/TimeElapsedDiagnosticsMiddleware.cs:31]:
```suggestion
                Log.Information("TimeElapsedDiagnosticsMiddleware - OK - CorrelationId: {CorrelationId}", correlationId);
```

File: src/WebApi/Logging/TimeElapsedDiagnosticsMiddleware.cs (hunk 1 above)
Comment (robertstriz) [reply src/WebApi/Logging/TimeElapsedDiagnosticsMiddleware.cs:47]: Dear @petrspelos 
Thank you for your valuable input.
I can not use Visual Studio. 
Best practice is to set rules in [.editorconfig](https://docs.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options?view=vs-2022) for other IDEs.
Could you please do that?

**Review by fpindej — APPROVED** [review]

File: src/WebApi/Logging/TimeElapsedDiagnosticsMiddleware.cs (hunk 1 above)
Comment (petrspelos) [reply src/WebApi/Logging/TimeElapsedDiagnosticsMiddleware.cs]: @PlesnikJakub

**Pull request merged**